app/controllers/admin/products_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Admin
+  class ProductsController < AdminController
+    def new
+      render :new, locals: { product: Product.new }
+    end
+
+    def create
+      product = Product.new(product_params)
+      if product.save
+        redirect_to root_path, notice: 'Product was successfully created'
+      else
+        flash.now[:alert] = 'Product was not created'
+        render :new, locals: { product: Product.new(product_params) }
```

Comment:
Wcześniej jak używałem zmiennych instancyjnych, to po prostu sobie tam były, ale teraz takie ręczne przekazywanie fajnie zmusza do zastanowienia się jak ten flow railsowy płynie

Reply:
nie no, to powinienes byl zrobic w tym PR. Wrzuciles w mastera kod, ktory nie dziala wiedzac o tym w momencie wrzucania ;) mozesz wczesniej przypisac do `product = Product.new(product_params)`. Potem gdy zrobisz `.save.` to ten obiekt bedzie miec przypisane bledy jakie wystapily i go zwroc na widok